experimental/Schemes/ToricIdealSheaves/constructors.jl

```diff
@@ -53,51 +53,56 @@ function IdealSheaf(X::NormalToricVariety, I::MPolyIdeal)
-    for m in hilbert_basis(weight_cone(U))
-      img = one(help_ring)
-      for j in 1:length(indices)
-        u_rho = matrix(ZZ,rays(X))[indices[j]:(indices[j]),:]
-        expo = (u_rho*m)[1]
-        img = img * x_rho[j]^expo
+  if is_smooth(X)
+    # We need to dehomogenize the ideal I in the Cox ring S to the local
+    # charts U_sigma, with sigma a cone in the fan of the variety.
+    # To this end we use Proposition 5.2.10 from Cox-Little-Schenck, p. 223ff.
+    # Abstractly, the chart U_sigma is isomorphic to C^n with n the number
+    # of ray generators of sigma, owing to the assumptions 1 an 2 above.
+    # Note however that the coordinates of C^n are not one to one to
+    # the homogeneous coordinates of the Cox ring. Rather, the coordinates 
+    # of the affine pieces correspond to the `hilbert_basis(polarize(sigma))`.
+
+    # TODO: In the long run we should think about making the creation of the 
+    # following dictionary lazy. But this requires partial rewriting of the 
+    # ideal sheaves as a whole, so we postpone it for the moment.
+
+    IM = maximal_cones(IncidenceMatrix, X)
+    for (k, U) in enumerate(affine_charts(X))
+
+      # We first create the morphism \pi_s* from p. 224, l. 3.
+      indices = [k for k in row(IM, k)]
+      help_ring, x_rho = polynomial_ring(QQ, ["x_$j" for j in indices])
+      imgs_phi_star = [j in indices ? x_rho[findfirst(k->k==j, indices)] : one(help_ring) for j in 1:n_rays(X)]
+      phi_s_star = hom(cox_ring(X), help_ring, imgs_phi_star)
+
+      # Now we need to create the inverse of alpha*.
+      imgs_alpha_star = elem_type(help_ring)[]
+      for m in hilbert_basis(weight_cone(U))
+        img = one(help_ring)
+        for j in 1:length(indices)
+          u_rho = matrix(ZZ,rays(X))[indices[j]:(indices[j]),:]
+          expo = (u_rho*m)[1]
+          img = img * x_rho[j]^expo
+        end
+        push!(imgs_alpha_star, img)
       end
-      push!(imgs_alpha_star, img)
-    end
-    alpha_star = hom(OO(U), help_ring, imgs_alpha_star)
+      alpha_star = hom(OO(U), help_ring, imgs_alpha_star)
 
-    # TODO: There should be better ways to create this map!
-    # Presumably, one can invert the matrix with the `expo`s as entries?
-    # @larskastner If you can confirm this, let's change this.
-    alpha_star_inv = inverse(alpha_star)
-    ideal_dict[U] = ideal(OO(U), [alpha_star_inv(phi_s_star(g)) for g in gens(I)])
+      # TODO: There should be better ways to create this map!
+      # Presumably, one can invert the matrix with the `expo`s as entries?
+      # @larskastner If you can confirm this, let's change this.
+      alpha_star_inv = inverse(alpha_star)
+      ideal_dict[U] = ideal(OO(U), [alpha_star_inv(phi_s_star(g)) for g in gens(I)])
+    end
+  else
+    for (k, U) in enumerate(affine_charts(X))
+      ideal_dict[U] = _dehomogenize_to_chart(X, I, k)
+    end
   end
 
   return IdealSheaf(X, ideal_dict, check=true) # TODO: Set the check to false eventually.
```

Comment:
Remember to switch off the tests here, eventually. 

After #3490 is merged, we can rebuild this work on the lazy sheaves.